include/git2/tag.h

```diff
@@ -365,6 +365,16 @@ GIT_EXTERN(int) git_tag_peel(
  */
 GIT_EXTERN(int) git_tag_dup(git_tag **out, git_tag *source);
 
+/**
+ * Determine whether a tag name is valid
+ *
+ * @param name A tag name to test
+ *
+ * @return 1 if tag name not valid, 0 if it is valid, an error code otherwise.
```

Comment:
`is_name_valid` is a boolean question.  We should return something truthy if it's true, and false if it's false.  

`if (!git_branch_is_name_valid()) { name is valid }` doesn't make sense.  Further, this matches `git_reference_is_name_valid`.

Reply:
Updated.